assets/src/edit-story/components/keyboardShortcutsMenu/shortcutMenu.js

```diff
@@ -17,153 +17,147 @@
-        ({ leftSideSections, rightSideSections }, section, index) => {
-          if (index < leftHalfCount) {
-            return {
-              leftSideSections: [...leftSideSections, section],
-              rightSideSections,
-            };
-          } else {
-            return {
-              leftSideSections,
-              rightSideSections: [...rightSideSections, section],
-            };
-          }
-        },
-        {
-          leftSideSections: [],
-          rightSideSections: [],
-        }
-      ),
-    [sections, leftHalfCount]
-  );
+const ScrollableContent = styled.div`
+  overflow: auto;
+  flex-shrink: 1;
+`;
 
-  const renderSection = useCallback(
-    ({ title, commands }) => (
-      <SectionWrapper key={title}>
-        <SectionHeader id={title}>{title}</SectionHeader>
-        <SectionContent role="group" aria-labelledby={title}>
-          {commands.map(({ label, shortcut }) => (
-            <Fragment key={label}>
-              <Label role="listitem">{label}</Label>
-              <ShortcutLabel
-                role="listitem"
-                keys={shortcut}
-                alignment={'left'}
-              />
-            </Fragment>
-          ))}
-        </SectionContent>
-      </SectionWrapper>
-    ),
-    []
-  );
+function ShortcutMenu({ toggleMenu }) {
+  const containerRef = useRef();
+  const closeRef = useRef();
+
+  const [bottomOffset, setBottomOffset] = useState(0);
+  useLayoutEffect(() => {
+    const node = containerRef.current;
+    if (!node) {
+      return undefined;
+    }
+    const calc = () => {
+      // Find node bottom offset from bottom of browser window
+      const nodeRect = node.getBoundingClientRect();
+      const docRect = node.ownerDocument.documentElement.getBoundingClientRect();
+      setBottomOffset(docRect.height - nodeRect.y - nodeRect.height);
+    };
+    // Call it now
+    calc();
+    // Call it in a frame
+    // Disable reason: No need to cancel this one
+    // eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+    setTimeout(calc);
+    // And call it every time the window resizes
+    window.addEventListener('resize', calc);
+    return () => window.removeEventListener('resize', calc);
+  }, []);
 
   const handleCloseClick = useCallback((e) => toggleMenu(e, false), [
     toggleMenu,
   ]);
 
   useEffect(() => {
-    closeRef.current.focus?.();
+    // When the popup opens, move focus inside it
+    // However, only move it to the close button if the user actually uses the keyboard
+    // otherwise the close button hover state will be triggered for a mouse user.
+    const targetRef = isKeyboardUser() ? closeRef : containerRef;
+    targetRef.current?.focus?.();
   }, []);
 
   useKeyDownEffect(containerRef, TOGGLE_SHORTCUTS_MENU, toggleMenu, [
     toggleMenu,
   ]);
 
+  useKeyDownEffect(containerRef, 'esc', toggleMenu);
+  useFocusTrapping({ ref: containerRef });
+
+  const headerLabelId = `kb-header-${uuidv4()}`;
```

Comment:
Are we worried about this generating a new id on every render?

I know technically this component _could_ be rendered multiple times on the DOM, but in actuality it's always going to be a singleton (given the context of designs/purpose). I'd be fine with just statically defining this outside the component vs adding the bloat of a memo & id generation to satisfy a use case we're never going to come across

Reply:
Yeah, it should be memoized, thanks. It's silly to regenerate - as the component re-renders every time the window resizes in height due to `bottomOffset` creating unnecessary DOM updates.

Reply:
And yes, memoizing is silly for this - moved to a const outside the component in 9bbfc914c.